certbot-nginx/certbot_nginx/_internal/nginxparser.py

```diff
@@ -138,100 +160,115 @@ def _coerce(self, inbound):
         :rtype: tuple
 
         """
-        if not isinstance(inbound, list):                      # str or None
+        if not isinstance(inbound, list):  # str or None
             return inbound, inbound
         else:
             if not hasattr(inbound, "spaced"):
                 inbound = UnspacedList(inbound)
             return inbound, inbound.spaced
 
-    def insert(self, i, x):
+    def insert(self, i: int, x: Any) -> None:
```

Comment:
I answer here for `insert`, `append` and `extend`. These three functions have their type dictated by the superclass (`List`) which says that the actual type of `x` must be the same than the generic of the `List`. So to make x be of type `Union[str, None, List[Any]]`, I need to make `UnspacedList` heritate from `List[Union[str, None, List[Any]]]` instead of `List[Any]`.

This generates around 50 errors in the type checking basically because of the recursive type problem we already discussed in https://github.com/certbot/certbot/pull/9124#discussion_r778453975. Quite similarly, we have to stick with `List[Any]`, which consequently leads `x` to be `Any` in `insert`, `append`  and `extend`.